keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -44,20 +58,9 @@ class ClusterHandler(
     "clusters"
   ) to Cluster::class.java
 
-  override fun validate(resource: Resource<*>, generateName: Boolean): Resource<Cluster> {
-    val cluster = objectMapper.convertValue<Cluster>(resource.spec)
-    var metadata = resource.metadata
-    if (generateName) {
-      val name = "ec2:cluster:${cluster.location.accountName}:${cluster.location.region}:${cluster.moniker.cluster}"
-      metadata = resource.metadata.copy(name = ResourceName(name))
-    }
-    return Resource(
-      apiVersion = resource.apiVersion,
-      kind = resource.kind,
-      metadata = metadata,
-      spec = cluster
-    )
-  }
+  override fun generateName(spec: Cluster) = ResourceName(
+    "ec2:cluster:${spec.location.accountName}:${spec.location.region}:${spec.moniker.cluster}"
+  )
```

Comment:
Now each handler just needs to implement a `generateName` and optionally a `validate`. I know we may be moving this logic to a separate type but it still makes sense to simplify it.

Reply:
Makes sense!